clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -130,6 +130,19 @@ public Thread newThread(Runnable runnable) {
         } else {
             this.metricsScheduler = null;
         }
+
+        this.sensor("metrics-count").add(new MetricName("count", "kafka-metrics-count", "total number of registered metrics", defaultConfig.tags()),
+            new MeasurableStat() {
+                @Override
+                public double measure(MetricConfig config, long now) {
+                    return metrics.size();
+                }
+
+                @Override
+                public void record(MetricConfig config, double value, long timeMs) {
+
+                }
+            });
```

Comment:
It seems there is no need to create a sensor here. You can just do

addMetric(new MetricName("count", "kafka-metrics-count", "total number of registered metrics", defaultConfig.tags(), new Measurable() {
                public double measure(MetricConfig config, long now) {
                    return metrics.size();
                }